core/conversion/converters/impl/conv_deconv.cpp

```diff
@@ -9,74 +9,147 @@ namespace conversion {
 namespace converters {
 namespace impl {
 namespace {
-auto conv_registrations TRTORCH_UNUSED = RegisterNodeConversionPatterns().pattern({
-    R"SIG(aten::_convolution(Tensor input, Tensor weight,
+auto conv_registrations TRTORCH_UNUSED =
+    RegisterNodeConversionPatterns()
+        .pattern({
+            R"SIG(aten::_convolution(Tensor input, Tensor weight,
                                  Tensor? bias, int[] stride, int[] padding,
                                  int[] dilation, bool transposed,
                                  int[] output_padding, int groups, bool benchmark,
-                                 bool deterministic, bool cudnn_enabled) -> (Tensor))SIG",
-    [](ConversionCtx* ctx, const torch::jit::Node* n, args& args) -> bool {
-      auto in = args[0].ITensor(); // assumes non-static input Tensor
-
-      auto w = Weights(ctx, args[1].unwrapToTensor());
-      auto stride = util::toDims(args[3].unwrapToIntList());
-      LOG_DEBUG("stride: " << stride);
-      auto padding = util::toDims(args[4].unwrapToIntList());
-      LOG_DEBUG("padding: " << padding);
-      auto dilation = util::toDims(args[5].unwrapToIntList());
-      LOG_DEBUG("dilation: " << dilation);
-      bool transposed = args[6].unwrapToBool();
-      auto out_padding = util::toDims(args[7].unwrapToIntList());
-      LOG_DEBUG("out_padding: " << out_padding);
-      int64_t groups = args[8].unwrapToInt();
-
-      nvinfer1::ILayer* new_layer;
-      if (transposed) {
-        nvinfer1::IDeconvolutionLayer* deconv;
-        if (args[2].IValue()->isTensor()) {
-          Weights b(ctx, args[2].IValue()->toTensor());
-          deconv = ctx->net->addDeconvolutionNd(*in, w.num_input_maps, w.kernel_shape, w.data, b.data);
-        } else {
-          deconv = ctx->net->addDeconvolutionNd(*in, w.num_input_maps, w.kernel_shape, w.data, {});
-        }
-
-        TRTORCH_CHECK(deconv, "Unable to create deconvolution layer from node: " << *n);
-
-        deconv->setStrideNd(stride);
-        deconv->setPaddingNd(padding);
+                                 bool deterministic, bool cudnn_enabled, bool allow_tf32) -> (Tensor))SIG",
```

Comment:
Are there specific things we need to do for tf32?

Reply:
On TRT side, we don't need to do anything. By default TF32 format is picked for layers which support them on Ampere GPUs. If we want to disable it, we need to add specific flag in TRT builder. I can add this in a separate PR.
On Pytorch side, if we export the torchscript model using these flags, 
```
torch.backends.cuda.matmul.allow_tf32 = True
torch.backends.cudnn.allow_tf32 = True
```
the graph will have this information on which format to use during pytorch execution.

Reply:
Can you file an issue to add an API to disable tf32, I think we want it on by default but should give users the option to turn it off if they want. Does not need to be in this release

Reply:
I didn't get a notification for your comment. Opened an issue now [https://github.com/NVIDIA/TRTorch/issues/247](url)